WordPress/src/main/java/org/wordpress/android/datasets/ReaderPostTable.java

```diff
@@ -1249,9 +1249,39 @@ private static void updateIsBookmarkedField(final ReaderPostList posts) {
         ReaderBlogIdPostIdList bookmarkedPosts = getBookmarkedPostIds();
         for (ReaderPost post : posts) {
             for (ReaderBlogIdPostId bookmarkedPostId : bookmarkedPosts) {
-                if (post.blogId == bookmarkedPostId.getBlogId() && post.postId == bookmarkedPostId.getPostId()) {
-                    post.isBookmarked = true;
-                }
+                if (isBookmarkedPost(post, bookmarkedPostId)) post.isBookmarked = true;
+            }
+        }
+    }
+
+    private static boolean isBookmarkedPost(ReaderPost post, ReaderBlogIdPostId bookmarkedPostId) {
+        return post.blogId == bookmarkedPostId.getBlogId() && post.postId == bookmarkedPostId.getPostId();
+    }
+
+    public static void updateBookmarkedPostsPseudoId(final ReaderPostList posts) {
+        ReaderBlogIdPostIdList bookmarkedPosts = getBookmarkedPostIds();
+        for (ReaderPost post : posts) {
+            for (ReaderBlogIdPostId bookmarkedPostId : bookmarkedPosts) {
+                if (isBookmarkedPost(post, bookmarkedPostId)) updateBookmarkedPostsPseudoId(post, bookmarkedPostId);
+            }
+        }
+    }
+
+    public static void updateBookmarkedPostsPseudoId(ReaderPost post, ReaderBlogIdPostId bookmarkedPostIds) {
```

Comment:
Minor (🔍): Consider renaming that to `bookmarkedPostId` instead of having it a plural, which might suggest that this is somehow a list.

PS: It is already a singular on the already existing `isBookmarkedPost(...)` method.

Reply:
Fixed in 0455b05